client/ayon_harmony/plugins/create/create_render_layers.py

```diff
@@ -0,0 +1,1014 @@
+                )
+            ]
+        )
+        return output
+
+    def product_impl(self, name, instance_data: dict, pre_create_data: dict):
+        pass
+
+    def _get_layer_positions_in_groups(self, layers):
+        layer_positions_in_groups = {}
+        sorted_layers = sorted(
+            layers, 
+            key=lambda layer: (layer["color"], layer["position"]),
+            reverse=True
+        )
+        last_group = None
+        position_in_group = 1
+        for layer in sorted_layers:
+            if last_group is None or last_group != layer["color"]:
+                position_in_group = 1
+                last_group = layer["color"]
+            layer_positions_in_groups[layer["name"]] = position_in_group
+            position_in_group += 1
+        return layer_positions_in_groups
+    
+    def _format_created_nodes(self, group_colors):
+        """Tries to wrap all nodes of a layer group into Backdrop"""
+        scene_containers = harmony.get_scene_data()
+        for node_name, container in scene_containers.items():
+            if container["creator_identifier"] != CreateRenderLayer.identifier:
+                continue
+            group_label = container["variant"]
+            group_id = container["creator_attributes"]["group_id"]
+            group_color = group_colors[group_id]
+            harmony.send(
+                {
+                    "function": f"AyonHarmony.Creators.CreateRenderLayer.formatNodes",  #noqa
+                    "args": [node_name, group_label, group_color]
+                }
+            )
+
+
+# TODO refactor
+def get_group_infos() -> list[GroupInfo]:
+    """Lists all used layer colors to choose from"""
+    # TODO cache this
+    used_group_colors = []
+    layers_data = get_layers_info()
+    # to keep order
+    available_colors = {layer["color"]:layer["color"] for layer in layers_data}
+
+    group_id = 1
+    for color in available_colors.keys():
+        item = GroupInfo(id=group_id, color=color)
+        used_group_colors.append(item)
+        group_id += 1
+
+    return used_group_colors
+
+
+def get_group_color(
+    group_id: int,
+    group_infos: Optional[list[GroupInfo]]=None
+) -> str:
+    """Find appropriate color for ordinal number of group"""
+    if not group_infos:
+        group_infos = get_group_infos()
+    group_color = None
+    for group_item in group_infos:
+        if group_item.id == group_id:
+            group_color = group_item.color
+            break
+    return group_color
+
+
+def get_group_id(
+    group_color: str,
+    group_infos: Optional[list[GroupInfo]]=None
+) -> int:
+    """Find ordinal number of group for particular group color"""
+    if not group_infos:
+        group_infos = get_group_infos()
+    group_id = None
+    for group_item in group_infos:
+        if group_item.color == group_color:
+            group_id = group_item.id
+            break
+    return group_id
+
+
+def get_group_layer_name(
```

Comment:
This should be either `get_render_layer_variant` or `get_group_name`. "group layer" is not a thing.